compiler/src/dotty/tools/dotc/typer/Namer.scala

```diff
@@ -360,7 +360,16 @@ class Namer { typer: Typer =>
   /** Expand tree and store in `expandedTree` */
   def expand(tree: Tree)(implicit ctx: Context): Unit = tree match {
     case mdef: DefTree =>
-      val expanded = desugar.defTree(mdef)
+      val expanded = mdef match {
+        case mdef: MemberDef if ctx.macrosEnabled && macros.hasAnnotMacros(mdef) =>
+          macros.expandAnnotMacro(mdef) match {
+            case Block(stats, _) =>                 // introducing of same level symbols
```

Comment:
I think it would be cleaner if `expandAnnotMacro` returned a `Thicket` directly. Why the detour via `Block`? What happens with its `expr` part?